R/secure_rmd.R

```diff
@@ -104,20 +107,42 @@ secure_rmd <- function(
   yaml_header <- yamlFromRmd(rmd_file_path)
 
   yaml_polished <- yaml_header$polished
-
-  if (is.null(yaml_polished)) {
-    stop('"polished" must be included in the YAML header', call. = FALSE)
-  }
-
-  if (is.null(yaml_polished$global_sessions_config$api_key)) {
-    yaml_polished$global_sessions_config$api_key <- get_api_key()
+  yaml_polished_global_config <- yaml_polished$global_sessions_config
+
+  # global_sessions_config_args overrides
+  # global_sessions_config_args
+  # remove any NULL
+  yaml_polished_global_config <- yaml_polished_global_config[
+    !sapply(yaml_polished_global_config, is.null)
+  ]
+  global_sessions_config_args <- global_sessions_config_args[
+    !sapply(global_sessions_config_args, is.null)
+  ]
+  inames <- intersect(names(global_sessions_config_args),
+                      names(yaml_polished_global_config))
+  if (length(inames) > 0) {
+    warning(
+      paste0(
+        paste0(inames, collapse = ", "),
+        " specified in ",
+        "global_sessions_config_args and YAML polished header,",
+        " using global_sessions_config_args"
+      )
+    )
+    yaml_polished_global_config <- yaml_polished_global_config[
+      !names(yaml_polished_global_config) %in% inames
+    ]
   }
 
   global_sessions_config_args <- modifyList(
     global_sessions_config_args,
```

Comment:
I'd probably switch the order of the lists passed to ModifyList because the values in the second list override values in the first list.  I know this isn't an issue because you remove the values from `yaml_polished_global_config` that are in both lists above, but I'd still switch them.